labs/b5.md

```diff
@@ -0,0 +1,364 @@
+
+Ports define a service endpoint, broadly speaking -- ports mark a point of traffic ingress and egress.  Ports are represented by a 16 bit number meaning thus ranging from 0 to 65535.  Ports from 0 to 1023 are well known ports, i.e. system ports.  Using these ports usually has a stricter requirement.  1024 to 49151 are registered ports. IANA maintains the official [list][port list] of well-known and registered ranges. The remaining ports from 49152 to 65535 are ephemeral ports which can be dynamically allocated for communication sessions on a per request basis.
+
+Some port numbers for well known services are as follows:
+Service     | Port
+------------|------
+HTTP        |80
+HTTPS       |443
+
+
+# Sysadmin Commands
+As a sysadmin, trying to diagnose network issues can often be pretty challenging. Given the scale and complexity of networks, it's tough trying to narrow down the scope of a problem to a point of failure. What follows is a list of commands/tools that can help with triaging problems. There are a lot of tools and we don't expect you to memorize every single detail. However, it is important to know what tools exist and when to use them when a problems inevitably arise. If you ever need more details the `man` pages for these commands are a great place to turn to for reference.
+
+Tools also tend to overlap in functionality -- for example there are multiple tools that can display interface information or test connectivity.  When possible, it is a good idea to use multiple tools to cross-check one another.
+
+Note that when it comes to real world networks there are even more factors to consider that we haven't touched on like network security. For example, two machines can have a fully functioning connection but if one machine has been configured to drop all packets then it might seem as if they aren't connected. 
+
+So take the output of these tools with a grain of salt, they a means of narrowing down issues. It is important not to misinterpret outputs or jump to conclusions too quickly.
+
+1. `hostname`
+A simple and straightforward command that can display information about a host, IP addresses, FQDN, and etc.
+
+2. `ping`
+ Another simple command, most of the time you'll be using ping as a first step towards testing connectivity. If ping can't reach a host then there is likely an issue with connectivity.
+
+    Moreover ping also provides metrics for Round Trip Time (RTT) and packet loss.  Round trip time is defined as the time it takes for a response to arrive after sending the ping packet.  These can prove to be very useful statistics.
+
+3. `traceroute`
+Traceroute sends packets Time to Live (TTL) equal to the number of hops. Routers decreases the value of TTL of an incoming packet and if it sees an incoming packet with TTL = 0 then drops it, otherwise it decreases the value and sends it further. At the same time it sends diagnosing information to the source about router's identity.
```

Comment:
I'd cut out this first paragraph, in favor of just linking an article, because the discussion of how exactly it works is likely to confuse students.

Reply:
I think the explanation here is broad enough, and having too many links results in them just being ignored.  If the students aren't satisfied with the given description, they can read the tools man page or google its internals